lib/commands/init.ts

```diff
@@ -0,0 +1,114 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import constants = require("./../constants");
+import helpers = require("./../common/helpers");
+import Future = require("fibers/future");
+import path = require("path");
+
+export class InitCommand implements ICommand {
+	private _projectFilePath: string;
+	
+	constructor(private $fs: IFileSystem,
+		private $errors: IErrors,
+		private $logger: ILogger,
+		private $options: IOptions,
+		private $injector: IInjector,
+		private $staticConfig: IStaticConfig,
+		private $projectHelper: IProjectHelper,
+		private $prompter: IPrompter,
+		private $npm: INodePackageManager,
+		private $npmInstallationManager: INpmInstallationManager) { }
+		
+	public allowedParameters: ICommandParameter[] = [];
+	public enableHooks = false;
+		
+	public execute(args: string[]): IFuture<void> {
+		return (() => {
+			let projectData: any = { };
+			
+			if(this.$fs.exists(this.projectFilePath).wait()) {
+				projectData = this.$fs.readJson(this.projectFilePath).wait();
+			}
+			
+			if(!projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE]) {
+				projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE] = { };
+			}
+			
+			let projectDataBackup = _.extend({}, projectData);
```

Comment:
This should be before
```TypeScript
if(!projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE]) {
```
At this point we might have changed the projectData, so this is not a real backup of the original data.